src/server/game/Entities/Player/CollectionMgr.cpp

```diff
@@ -259,3 +260,137 @@ bool CollectionMgr::CanApplyHeirloomXpBonus(uint32 itemId, uint32 level)
 
     return level <= 60;
 }
+
+MountDefinitionMap CollectionMgr::_mountDefinitions;
+
+void CollectionMgr::LoadAccountMounts(PreparedQueryResult result)
+{
+    if (!result)
+        return;
+
+    do
+    {
+        Field* fields = result->Fetch();
+        uint32 spellId = fields[0].GetUInt32();
+        bool isFavourite = fields[1].GetBool();
+
+        if (!sDB2Manager.GetMount(spellId))
+            continue;
+
+        _mounts[spellId] = isFavourite;
+    } while (result->NextRow());
+}
+
+void CollectionMgr::SaveAccountMounts(SQLTransaction& trans)
+{
+    PreparedStatement* stmt = nullptr;
+    for (auto const& mount : _mounts)
+    {
+        stmt = LoginDatabase.GetPreparedStatement(LOGIN_REP_ACCOUNT_MOUNTS);
+        stmt->setUInt32(0, _owner->GetBattlenetAccountId());
+        stmt->setUInt32(1, mount.first);
+        stmt->setBool(2, mount.second);
+        trans->Append(stmt);
+    }
+}
+
+bool CollectionMgr::UpdateAccountMounts(uint32 spellId, bool isFavourite)
+{
+    return _mounts.insert(MountContainer::value_type(spellId, isFavourite)).second;
+}
+
+void CollectionMgr::LoadMounts()
+{
+    for (auto const& m : _mounts)
+        AddMount(m.first, m.second);
+}
+
+bool CollectionMgr::AddMount(uint32 spellId, bool isFavourite /*= false*/, bool factionMount)
+{
+    Player* player = _owner->GetPlayer();
+    if (!player)
+        return false;
+
+    MountEntry const* mount = sDB2Manager.GetMount(spellId);
+    if (!mount)
+        return false;  
+
+    MountDefinitionMap::const_iterator itr = _mountDefinitions.find(spellId);
+    if (itr != _mountDefinitions.end() && !factionMount)
+        AddMount(itr->second, isFavourite, true);
+
+    if (mount->PlayerConditionId)
+    {
+        PlayerConditionEntry const* playerCondition = sPlayerConditionStore.LookupEntry(mount->PlayerConditionId);
+        if (!ConditionMgr::IsPlayerMeetingCondition(player, playerCondition))
+            return false;
+    }
+
+    UpdateAccountMounts(spellId, isFavourite);
+    player->LearnSpell(spellId, false);
+
+    return true;
+}
+
+void CollectionMgr::MountSetFavorite(uint32 spellId, bool favorite)
+{
+    MountContainer::iterator itr = _mounts.find(spellId);
+    if (itr == _mounts.end()) 
+        return;
+
+    Player* player = _owner->GetPlayer();
+    if (!player)
+        return;
+
+    itr->second = favorite;
+
+    // Temporary container, just need to store only selected mount
+    MountContainer tempMounts;
+    tempMounts.insert(*itr);
+    
+    WorldPackets::Misc::AccountMountUpdate mountUpdate;
+    mountUpdate.IsFullUpdate = false;
```

Comment:
maybe you need to send AccountMountUpdate inside addmount, because the only thing that I see fixes the issue at relog is that update when added to map